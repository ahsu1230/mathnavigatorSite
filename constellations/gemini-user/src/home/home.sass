@import "../utils/_constants.sass"

#home-banner-container
  position: relative
  width: 100%
  height: 500px

  & #banner-bg-img
    width: 100%
    height: 100%
    background: url('../../assets/sincos.jpg')
    background-size: cover
    background-position: center

  & #banner-bg-overlay
    position: absolute
    top: 0
    left: 0
    width: 100%
    height: 100%
    background-color: $colorTurquoiseDark
    opacity: 0.6

  & #banner-content
    position: absolute
    top: 40%
    left: 10%
    transform: translateY(-50%)

    & h1
      color: $colorWhite
      font-size: $fontSize4
      line-height: $lineHeight4

      @media screen and (max-width: $mediaWidth1)
        font-size: $fontSize2

    & h2
      margin-bottom: $spacing2x
      color: $colorTurquoiseLight
      font-size: $fontSize0
      line-height: $lineHeight0


  & button
    position: absolute
    right: 10%
    bottom: 20%
    padding: $spacing2x $spacing3x

#view-home .section
  width: $containerWidthPct
  max-width: $containerWidthPx1
  margin: 0 auto
  background-color: $colorWhite

  & h2
    margin: $spacing16x 0
    font-family: $fontFamilyRockwell
    font-size: $fontSize2
    line-height: $lineHeight2
    letter-spacing: $letterSpacingRoboto
    text-align: center


// Home Tile Cards
.card-container

  & + div
    text-align: center
    margin-top: $spacing12x
    margin-bottom: $spacing16x

.home-tile-card
  display: inline-block
  width: calc(33% - 46px)
  padding: 0 $spacing3x
  vertical-align: top

  &:first-child
    margin-left: 0
  &:last-child
    margin-right: 0

  & .img-container
    position: relative
    height: $spacing24x

    & img
      position: relative
      left: 50%
      transform: translateX(-50%)
      width: 100%
      max-width: $spacing24x

  & h3
    color: $colorPrimary
    margin: $spacing4x 0
    font-size: $fontSize2
    line-height: $lineHeight1
    text-align: center

  & p
    margin-bottom: $spacing4x
    color: $colorBlackDark
    text-align: center

  & button
    position: relative
    left: 50%
    transform: translateX(-50%)


// Home Section - Programs
#view-home .section.programs
  & .home-tile-card
    @media screen and (max-width: $mediaWidth2)
      display: block
      width: 100%
      margin-bottom: $spacing8x
      box-sizing: border-box

      & p
        width: $spacing32x
        margin: 0 auto $spacing4x

      & .img-container
        max-height: $spacing16x


// Home Section - Success
#view-home .section.success
  border-top: 1px solid $colorGrayLight

  & .img-container
    background-color: $colorPrimary
    height: $spacing24x

    & img
      top: 50%
      left: 50%
      transform: translate(-50%, -50%)
      width: $spacing12x

  & h3
    padding: $spacing2x 0
    max-height: $spacing16x

  & .home-tile-card
    @media screen and (max-width: $mediaWidth2)
      display: block
      width: 60%
      min-width: $spacing40x
      margin: 0 auto $spacing8x
      box-sizing: border-box

      & p
        width: $spacing32x
        margin: 0 auto $spacing4x

      & .img-container
        width: $spacing32x
        margin: 0 auto


// Home Section - Stories
#view-home .section.stories
  position: relative
  width: 100%
  max-width: 800px
  margin: $spacing12x auto $spacing24x

  & .story-banner-container
    position: relative

    & .story-banner-img
      width: 100%
      height: $spacing48x
      background: url('../../assets/classroom1.jpg')
      background-position: center
      background-size: cover

    & .story-banner-overlay
      position: absolute
      top: 0
      left: 0
      width: 100%
      height: 100%
      background: $colorTurquoiseDark
      opacity: 0.2

    & .text-container
      position: absolute
      top: 50%
      left: 50%
      transform: translate(-50%, -50%)
      padding: $spacing4x $spacing8x
      background: rgba(0, 0, 0, 0.7)

      & h2
        margin: 0 0 $spacing4x
        color: $colorWhite

    & .action-join
      position: absolute
      right: 0
      bottom: 0
