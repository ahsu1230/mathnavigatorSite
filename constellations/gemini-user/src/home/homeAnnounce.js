"use strict";
require("./homeAnnounce.sass");
import React from "react";
import { Link } from "react-router-dom";
import { isEmpty, filter } from "lodash";
import srcClose from "../../assets/close_white.svg";
import API from "../utils/api.js";

export class HomeAnnounce extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            show: false,
            targetAnnounce: {},
        };
    }

    componentDidMount() {
        console.log("api attempt ");
        API.get("api/announcements/all").then((res) => {
            const announceList = res.data;
            console.log("api success!");

            let valid = filter(announceList, (a) => a.onHomePage);
            let targetAnnounce = valid.length > 0 ? valid[0] : undefined;

            targetAnnounce.shortMessage = shortenMessage(
                targetAnnounce.message
            );

            this.showTimeout = setTimeout(() => {
                this.setState({ show: true });
            }, 1000);

            this.setState({
                targetAnnounce: targetAnnounce,
            });
        });
    }

    dismiss = () => {
        this.setState({ show: false });
    };

    render() {
        const announce = this.state.targetAnnounce;
        var showAnnounce = !isEmpty(announce);
        var component;
        if (showAnnounce) {
            component = (
                <Popup
                    announce={announce}
                    show={this.state.show}
                    announceHeight={this.state.announceHeight}
                    dismiss={this.dismiss}
                />
            );
        } else {
            component = <div></div>;
        }
        return component;
    }
}

class Popup extends React.Component {
    render() {
        const announce = this.props.announce;
        const show = this.props.show;
        const announceClass = show ? "show" : "";
        return (
            <div key="real" id="home-announce" className={announceClass}>
                <div className="header-container">
                    <h3>New Announcement!</h3>
                    <button className="close-x" onClick={this.props.dismiss}>
                        <div className="img-container">
                            <img src={srcClose} />
                        </div>
                    </button>
                </div>
                <div className="text-container">
                    <p>{announce.shortMessage}</p>
                </div>
                <Link to="/announcements" onClick={this.props.dismiss}>
                    Read more &#62;
                </Link>
            </div>
        );
    }
}

function shortenMessage(message) {
    var array = message.split(" ");
    var shortMessage = "";
    var needEllipsis = false;

    var i = 0;
    while (i < array.length) {
        var append = shortMessage + array[i];
        if (append.length > 120) {
            needEllipsis = true;
            break;
        }
        shortMessage += array[i] + " ";
        i++;
    }

    if (needEllipsis) {
        shortMessage = shortMessage.slice(0, -1); // remove last "space"
        shortMessage += "...";
    }
    return shortMessage;
}
