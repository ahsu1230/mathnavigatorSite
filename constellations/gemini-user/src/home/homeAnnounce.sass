@import "../utils/_constants.sass"

#home-announce
    position: fixed
    bottom: $spacing4x
    left: $spacing4x
    transform: translate(-20%, 20%) scale(0.7)
    width: $spacing48x
    background-color: $colorWhite
    border-radius: 12px 12px 12px 0px
    box-shadow: 4px 4px 12px 4px rgba(0, 0, 0, 0.4)
    overflow: hidden
    visibility: hidden
    opacity: 0.0
    transition: all linear 0.15s

    @media screen and (max-width: $containerWidthPxTablet)
        transform: translate(0, 20%)
        border-radius: 0px

    &.show
        transform: translate(0, 0) scale(1.0)
        visibility: visible
        opacity: 1.0

    @media screen and (max-width: $containerWidthPxTablet)
        bottom: 0
        left: 0
        width: 100%

    .header-container
        display: flex
        flex-direction: row
        justify-content: space-between
        align-items: center
        background-color: $colorTurquoiseLight
        color: $colorWhite

        h3
            padding: $spacing1x 0 $spacing1x $spacing2x
            line-height: $lineHeight2
            font-size: $fontSize2
            font-weight: 400
            color: $colorWhite

        button.close-x
            width: 32px
            height: 32px
            padding: 0
            margin-right: $spacing1x
            background-color: transparent

            &:hover
                .img-container
                    background-color: $colorTurquoise
                img
                    transform: scale(1.1)

            .img-container
                width: 100%
                height: 100%
                padding: 6px
                border-radius: 50%
                box-sizing: border-box

            img
                width: 100%
                height: 100%
    
    .text-container
        padding: 0 $spacing2x
        line-height: $lineHeight1
        font-size: $fontSize1

    a
        float: right
        padding-right: $spacing2x
        padding-bottom: $spacing2x
        line-height: $lineHeight1
        font-size: $fontSize1
        color: $colorTurquoise
        

