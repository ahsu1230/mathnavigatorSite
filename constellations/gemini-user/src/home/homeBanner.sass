@import "../utils/_constants.sass"

#home-banner-container
  position: relative
  width: 100%
  height: $spacing72x

  #banner-bg-img
    width: 100%
    height: 100%
    background: url('../../assets/banner_compass.jpg')
    background-size: cover
    background-position: center
    filter: blur(2px)
    -webkit-filter: blur(2px)

  #banner-bg-overlay
    position: absolute
    top: 0
    left: 0
    width: 100%
    height: 100%
    background-color: $colorBlack
    opacity: 0.7

  #banner-content
    display: flex
    flex-direction: row
    justify-content: space-between
    width: $spacing96x
    box-sizing: border-box
    position: absolute
    top: 45%
    left: 50%
    transform: translate(-50%, -50%)
    color: $colorWhite

    .texts
      text-align: left

      h1
        letter-spacing: 1.5px

    .buttons
      display: flex
      flex-direction: column
      padding-top: $spacing8x
      text-align: right

      button
        border-radius: 4px

        &.enroll
          margin-bottom: $spacing4x
        &.browse
          border: 1px solid $colorWhite
          color: $colorWhite
          background-color: transparent

          &:hover
            background-color: $colorWhite
            color: $colorTurquoise


@media screen and (max-width: $mediaWidth2)
  #home-banner-container #banner-content
    flex-direction: column
    width: 100%

    h1
      font-size: $fontSize1
      margin-bottom: $spacing4x
    p
      font-size: $fontSize0
      margin: $spacing1x 0

    .texts
      text-align: center
    .buttons
      text-align: center
      padding-top: $spacing8x