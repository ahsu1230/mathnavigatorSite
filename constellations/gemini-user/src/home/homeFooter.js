"use strict";
require("./homeFooter.sass");
import React from "react";
import { Link } from "react-router-dom";

export default class HomeSectionFooter extends React.Component {
    render() {
        return (
            <div className="section home-footer">
                <div className="content">
                    <h1>Try your first class with Math Navigator today!</h1>
                    <Link to="/enroll">
                        <button>Get Started</button>
                    </Link>
                </div>
            </div>
        );
    }
}
