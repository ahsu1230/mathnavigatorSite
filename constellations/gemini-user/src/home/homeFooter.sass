@import "../utils/_constants.sass"

// Home Section - Footer
#view-home .section.home-footer
    width: 100%
    max-width: unset
    height: $spacing48x
    margin: 0
    background-color: $colorTurquoiseLight
    box-sizing: border-box

    .content
        position: relative
        top: 50%
        left: 50%
        transform: translate(-50%, -50%)

    h1
        margin: 0 auto
        color: $colorWhite
        text-align: center

        @media screen and (max-width: $mediaWidth1)
            max-width: $spacing40x

    button
        display: block
        margin: $spacing4x auto
        padding: $spacing2x $spacing4x
        border-radius: 8px
        background-color: $colorOrange

    a
        color: $colorWhite
        &:hover
            color: $colorWhite
            text-decoration: none