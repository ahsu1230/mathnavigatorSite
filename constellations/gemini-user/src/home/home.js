"use strict";
require("./home.sass");
import React from "react";
import HomeBanner from "./homeBanner.js";
import HomeSectionPrograms from "./homePrograms.js";
import HomeSectionStrategy from "./homeStrategy.js";
import HomeSectionFooter from "./homeFooter.js";
import HomeSectionOnline from "./homeOnline.js";
import { HomeAnnounce } from "./homeAnnounce.js";

export class HomePage extends React.Component {
    render() {
        return (
            <div id="view-home">
                <div id="view-home-container">
                    <HomeBanner />
                    <HomeSectionPrograms />
                    <HomeSectionStrategy />
                    <HomeSectionOnline />
                    <HomeSectionFooter />
                    <HomeAnnounce />
                </div>
            </div>
        );
    }
}
