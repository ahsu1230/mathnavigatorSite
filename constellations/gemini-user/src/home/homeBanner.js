"use strict";
require("./home.sass");
require("./homeBanner.sass");
import React from "react";
import { Link } from "react-router-dom";

export default class HomeBanner extends React.Component {
    render() {
        return (
            <div id="home-banner-container">
                <div id="banner-bg-img"></div>
                <div id="banner-bg-overlay"></div>
                <div id="banner-content">
                    <div className="texts">
                        <h1>Navigate to success with Math Navigator</h1>
                        <p>
                            Providing affordable, high quality education
                            <br />
                            to thousands of students for 15 years.
                        </p>
                        <p>Montgomery County, MD</p>
                    </div>
                    <div className="buttons">
                        <Link to="/register">
                            <button className="enroll">
                                Enroll into a Class
                            </button>
                        </Link>
                        <Link to="/programs">
                            <button className="browse">
                                Browse Program Catalog
                            </button>
                        </Link>
                    </div>
                </div>
            </div>
        );
    }
}
