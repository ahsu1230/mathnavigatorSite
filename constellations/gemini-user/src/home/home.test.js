import React from "react";
import Enzyme, { shallow } from "enzyme";
import { HashRouter as Router } from "react-router-dom";
import { HomePage } from "./home.js";
import renderer from "react-test-renderer";

describe("test", () => {
    const component = shallow(<HomePage />);

    test("renders", () => {
        expect(component.exists()).toBe(true);
        expect(component.exists("HomeAnnounce"));
        expect(component.exists("HomeBanner"));
        expect(component.exists("HomeSectionPrograms"));
        expect(component.exists("HomeSectionSuccess"));
        expect(component.exists("HomeSectionStories"));
    });

    test("snapshot", () => {
        const tree = renderer
            .create(
                <Router>
                    <HomePage />
                </Router>
            )
            .toJSON();
        //expect(tree).toMatchSnapshot();
    });
});
