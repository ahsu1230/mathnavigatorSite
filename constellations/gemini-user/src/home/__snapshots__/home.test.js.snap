// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div
  id="view-home"
>
  <div
    id="view-home-container"
  >
    <div
      id="home-banner-container"
    >
      <div
        id="banner-bg-img"
      />
      <div
        id="banner-bg-overlay"
      />
      <div
        id="banner-content"
      >
        <div
          className="texts"
        >
          <h1>
            Navigate to success with Math Navigator
          </h1>
          <p>
            Providing affordable, high quality education
            <br />
            to thousands of students for 15 years.
          </p>
          <p>
            Montgomery County, MD
          </p>
        </div>
        <div
          className="buttons"
        >
          <a
            href="#/register"
            onClick={[Function]}
          >
            <button
              className="enroll"
            >
              Enroll into a Class
            </button>
          </a>
          <a
            href="#/programs"
            onClick={[Function]}
          >
            <button
              className="browse"
            >
              Browse Program Catalog
            </button>
          </a>
        </div>
      </div>
    </div>
    <div
      className="section programs"
    >
      <h2>
        Our Featured Programs
      </h2>
      <div
        className="card-container"
      >
        <div
          className="card"
        >
          <div
            className="img-container"
          >
            <img
              src="http://d2dwqi4dzedhxu.cloudfront.net/images/cb_sat.jpg"
            />
          </div>
          <h3>
            SAT 1 & 2
          </h3>
          <p>
            Our most popular programs. We provide dedicated courses for both the SAT1 Math Section and SAT Subject Test (Math Level 2).
          </p>
          <a
            href="#/programs"
            onClick={[Function]}
          >
            <button>
              Enroll Now
            </button>
          </a>
        </div>
        <div
          className="card"
        >
          <div
            className="img-container"
          >
            <img
              src="http://d2dwqi4dzedhxu.cloudfront.net/images/ap_calc.jpg"
            />
          </div>
          <h3>
            AP Calculus
          </h3>
          <p>
            Get college credits early by acing the AP Calculus Exams. This program is dedicated to both Calculus AB and BC students.
          </p>
          <a
            href="#/programs"
            onClick={[Function]}
          >
            <button>
              Enroll Now
            </button>
          </a>
        </div>
        <div
          className="card"
        >
          <div
            className="img-container"
          >
            <img
              src="http://d2dwqi4dzedhxu.cloudfront.net/images/amc.jpg"
            />
          </div>
          <h3>
            AMC
          </h3>
          <p>
            Administered by the Mathematics Association of America. Represent your school and compete with other mathematics athletes.
          </p>
          <a
            href="#/programs"
            onClick={[Function]}
          >
            <button>
              Enroll Now
            </button>
          </a>
        </div>
      </div>
      <a
        className="link-bar"
        href="#/programs"
        onClick={[Function]}
      >
        Visit our 
        <b>
          Program Catalog
        </b>
         to view more programs to enroll into.
      </a>
    </div>
    <div
      className="section strategy"
    >
      <h2>
        Navigate to Success
      </h2>
      <div
        className="card-container"
      >
        <div
          className="card"
        >
          <div
            className="img-container"
          >
            <img
              src="test-file-stub"
            />
          </div>
          <h3>
            Dedicated Attention
          </h3>
          <p>
            We dedicate time to each individual student and provide plans to improve their understanding of topics.
          </p>
        </div>
        <div
          className="card"
        >
          <div
            className="img-container"
          >
            <img
              src="test-file-stub"
            />
          </div>
          <h3>
            Strategic Thinking
          </h3>
          <p>
            We want our students to work smart, not just hard. With Andy's problem solving strategies, students will learn how to tackle any problem.
          </p>
        </div>
        <div
          className="card"
        >
          <div
            className="img-container"
          >
            <img
              src="test-file-stub"
            />
          </div>
          <h3>
            Abundant Resources
          </h3>
          <p>
            Andy makes all of the teaching material. With our huge collection of resources, students will go into their exams full of confidence.
          </p>
        </div>
      </div>
    </div>
    <div
      className="section online"
    >
      <h2>
        Our Online-only Solution
      </h2>
      <div
        className="content"
      >
        <img
          src="http://d2dwqi4dzedhxu.cloudfront.net/images/zoom_classroom.jpg"
        />
        <p>
          With the impact of Covid-19 and the community mandate to practice safe social distancing, all Math Navigator courses have been moved online!
          <br />
          <br />
          Using Google Classroom to manage classwork and Zoom video conferencing, teachers and students can discuss and share materials with ease. Despite the inconvenience of schools and community centers closing down, we are still dedicated to providing students with superior education.
        </p>
      </div>
    </div>
    <div
      className="section home-footer"
    >
      <div
        className="content"
      >
        <h1>
          Try your first class with Math Navigator today!
        </h1>
        <a
          href="#/register"
          onClick={[Function]}
        >
          <button>
            Get Started
          </button>
        </a>
      </div>
    </div>
    <div />
  </div>
</div>
`;
